### ISSUE TITLE
Unable to use GraphQL enum in error extension

### ISSUE THREAD

**Issue by mattoni** [async-graphql/async-graphql#617]
## Expected Behavior
I should be able to return an enum as part of an error extension

## Actual Behavior

The following error occurs on `e.set(Test::A)` where Test is an enum defined as: 

```
#[derive(Enum, Copy, Clone, Eq, PartialEq)]
pub enum Test {
    A,
    B,
    C,
}
```

and the implementation is: 

```
impl ErrorExtensions for AuthError {
    fn extend(&self) -> Error {
        Error::new(format!("{}", self)).extend_with(|err, e| match self {
            AuthError::MissingCapabilities => {
                e.set("code", "AUTH_MISSING_CAPS");
                e.set("capabilities", Test::A);
            }
        })
    }
}
```

```
the trait bound `async_graphql::Value: std::convert::From<graphql::auth::error::Test>` is not satisfied
the following implementations were found:
  <async_graphql::Value as std::convert::From<&'a [T]>>
  <async_graphql::Value as std::convert::From<&'a str>>
  <async_graphql::Value as std::convert::From<()>>
  <async_graphql::Value as std::convert::From<async_graphql::Variables>>
and 17 others
required because of the requirements on the impl of `std::convert::Into<async_graphql::Value>` for `graphql::auth::error::Test`
```

## Specifications

- Version: 2.9.5
- Platform: MacOS
- Subsystem:

**Comment by Miaxos** [comment]
Hum weird, there is a missing Form impl somewhere...

While resolving the issue, you can put `Test::A.to_value()` it'll work.

**Comment by sunli829** [comment]
This is not a bug, I didn't expect anyone to use it like this, but it looks good, I will support this feature later.😃

**Comment by sunli829** [comment]
I have added this feature to the `master` branch.

**Comment by sunli829** [comment]
Released in `v2.10.0`
